src/python/pants/backend/jvm/tasks/scalafmt.py

```diff
@@ -18,20 +84,22 @@ class ScalaFmt(RewriteBase):
   process_result to run different scalafmt commands.
   """
 
+  def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+    self._all_command_lines = []
```

Comment:
Rather than modifying the actual production code to also act like a spy, could we in the test code use a fake binary for scalafmt which logs its command line or something? Or unit test the thread-splitting logic specifically?